src/java.base/share/native/libjava/jni_util.h

```diff
@@ -235,6 +235,12 @@ JNU_GetStaticFieldByName(JNIEnv *env,
  * Miscellaneous utilities used by the class libraries
  */
 
+#define RESTARTABLE(_cmd, _result) do { \
+  do { \
+    _result = _cmd; \
+  } while((_result == -1) && (errno == EINTR)); \
+} while(0)
+
```

Comment:
jni_util.h is for all platforms so not the right place for a Unix specific macro. We need think where the right place for this, might have to introduce a jni_util_md.h.

Reply:
We could maybe also use the existing  https://github.com/openjdk/jdk/blob/master/src/java.base/unix/native/include/jni_md.h  - what do you think ?

Reply:
> We could maybe also use the existing https://github.com/openjdk/jdk/blob/master/src/java.base/unix/native/include/jni_md.h - what do you think ?

jni_md.h goes into the JDK run-time image (for jni.h to include) so I don't think we should be changing that.  jni_util.* is JDK internal so I think we're just missing a platform dependent include file.